src/SelfUpdateCommand.php

```diff
@@ -101,38 +127,54 @@ protected function getReleasesFromGithub()
     }
 
     /**
-     * Get latest release according to given constraints
+     * Get the latest release version and download URL according to given constraints.
+     *
+     * @throws \Exception
+     *
+     * @return string[]|null
+     *  "version" and "download_url" elements if the latest release is available, otherwise - NULL.
      */
-    public function getLatestReleaseFromGithub($preview = false, $major_constraint = '') {
-        $releases = $this->getReleasesFromGithub();
-        $version = null;
-        $url = null;
-
-        foreach ($releases as $release) {
+    public function getLatestReleaseFromGithub() {
```

Comment:
Stylistically I prefer passing local variables to using class fields, as it makes methods easier to understand. The best method vis-a-vis testability is one that takes parameters and returns a result, with no side effects. This is a good design aesthetic even when we're not providing unit tests yet. Optional.

Reply:
TBH, was hesitating about this approach initially, refactored to get rid of class fields.